scripts/tcs_startpool.sh

```diff
@@ -0,0 +1,311 @@
+        echo "--------Reseting the config file-----------"
+        mv $TCF_HOME/config/wpe_config.toml{.b,}
+    fi
+
+    # Changing library name in wpe_config.toml file to support mutiple Workerpool
+    # Saving the original file with .b extension and rewriting the new library name 
+    echo "Writing to config file"
+    cp $TCF_HOME/config/wpe_config.toml $TCF_HOME/config/wpe_config.toml.b
+    # Check line by line and replace enclave lib file name appended with interger COUNT
+    while read a; do
+        echo ${a//libavalon-wpe-enclave.signed.so/libavalon-wpe-enclave-$COUNT.signed.so}
+    done < $TCF_HOME/config/wpe_config.toml > $TCF_HOME/config/wpe_config.toml.t 
+    mv $TCF_HOME/config/wpe_config.toml{.t,}
+
+    # Building the code 
+    cd $TCF_HOME/tools/build
+    export ENCLAVE_TYPE=wpe
+    export WORKLOADS=$1
+    echo "Building WPE with WORKLOADS: $1 "
+    make  &> /dev/null
+    
+    # Renaming the enclave lib file as per the wpe_config.toml file 
+    echo "Renaming to lib file name"
+    mv $TCF_HOME/tc/sgx/trusted_worker_manager/enclave/build/lib/libavalon-wpe-enclave{,-$COUNT}.signed.so
+    mv $TCF_HOME/tc/sgx/trusted_worker_manager/enclave/build/lib/libavalon-wpe-enclave{,-$COUNT}.so
+    mv $TCF_HOME/tc/sgx/trusted_worker_manager/enclave/build/lib/libavalon-wpe-enclave{,-$COUNT}.signed.so.meta
+    
+    # Increment the count for next workerpool
+    ((COUNT++))
+}
+
+reset_config_file()
+{
+    echo "--------Reseting the config file-----------"
+    mv $TCF_HOME/config/wpe_config.toml{.b,} &> /dev/null
+    sleep 10
+}
+
+check_file()
+{
+    while(true)
+    do
+        FILE=$TCF_HOME/wpe_mr_enclave.txt
+        if test -f "$FILE"; then
+            echo "------------------------------------------------$FILE exists."
+            cat $TCF_HOME/wpe_mr_enclave.txt
+            return
+        fi  
+    done
+}
+
+stop_avalon_components()
+{
+    for i in $COMPONENTS ; do
+        pkill -f "$i"
+    done
+    echo "Hyperledger Avalon successfully ended."
+    pkill -f "$ENCLAVE_MANAGER_KME"
+    pkill -f "$ENCLAVE_MANAGER_WPE"
+    if [ $START_STOP_AVALON_SERVICES = 1 ] ; then
+        pkill -f "$KV_STORAGE"
+        pkill -f "$LISTENER"
+    fi
+    mv $TCF_HOME/config/wpe_config.toml{.b,} &> /dev/null
+    exit
+}
+
+stop_avalon_components_forcefully()
+{
+    ps -ef | grep bin/$LISTENER | grep -v grep | awk '{print $2}' | xargs -r kill -9 ;
+    ps -ef | grep bin/$KV_STORAGE | grep -v grep | awk '{print $2}' | xargs -r kill -9;
+    ps -ef | grep $ENCLAVE_MANAGER_KME | grep -v grep | awk '{print $2}' | xargs -r kill -9;
+    ps -ef | grep $ENCLAVE_MANAGER_WPE | grep -v grep | awk '{print $2}' | xargs -r kill -9;
+
+    allPorts=("bind zmq_url remote_storage_url")
+    for i in $allPorts ; do
+	 # Port number of listerner, zmq and kv storage is picked from listener toml file.
+         # grep command reads the line as string from toml file which contails url.eg: bind = "http://localhost:1947".
```

Comment:
s/contails url.eg/contains the url. E.g./

Reply:
done